Added Cdata class to allow writing of CDATA values

zyura opened the pull request:
Sometimes it's desired to use CDATA in the resulting XML file instead of regular XML escaping.

Cdata wrapper class allows to achieve this:

```php
$html_desc = '<p>Some HTML code</p>';
$offer = (new OfferSimple())
    ->setDescription(new Cdata($html_desc))
```

will result in
```xml
<description><![CDATA[<p>Some HTML code</p>]]></description>
```

instead of

```xml
<description>&lt;p&gt;Some HTML code&lt;/p&gt;</description>
```

Bukashk0zzz commented:
Hi, Thank you for PR.

There are some code style problems https://travis-ci.org/Bukashk0zzz/YmlGenerator/jobs/516475287

Can you please fix them before I merge this?

zyura commented:
Sure, I'll commit the fixed version soon.

zyura commented:
Fixed all code style issues, added a test for the Cdata class: OfferCdataGeneratorTest.
Had to raise method visibility of AbstractGeneratorTest::createOffers() to protected because it didn't allow to define custom offer description: description set in overridden createOffer() was overwritten by
```php
    ->setDescription($this->faker->sentence)
```
